### PR TITLE
Update Python dependencies

### PR DESCRIPTION
**Pull request by davidfischer**
- Numpy added some backward incompatible changes in v2
- The really old allauth we are using had some incompatibilities with newer setuptools

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
 
 # Form helpers
-crispy-bootstrap4==2022.1
-django-crispy-forms==2.0
+crispy-bootstrap4==2024.1
+django-crispy-forms==2.2
 
 # Authentication
-django-allauth==0.49.0
+django-allauth==0.63.3
```
Comment (ericholscher) [inline requirements/base.txt:26]: I know we hit some various issues with upgrades here on RTD, but we're doing way more complex stuff there, so hopefully it'll just work.

**Review by ericholscher — APPROVED** [review]

**Comment by ericholscher** [comment]
I'd love to see us using pip-compile similar to RTD on these deps, which makes updating this stuff pretty simple: https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html#pin-your-transitive-dependencies

**Comment by davidfischer** [comment]
> I'd love to see us using pip-compile similar to RTD on these deps, which makes updating this stuff pretty simple

I agree. We should switch. I don't know it at all but I guess it's time to learn.

**Comment by ericholscher** [comment]
I can put together a PR with it :+1:

**Comment by ericholscher** [comment]
https://github.com/readthedocs/ethical-ad-server/pull/884

**Pull request merged**